sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageProcessor.java

```diff
@@ -22,45 +27,152 @@
 
     @Value("${PREFETCH_COUNT:0}")
     private int prefetchCount;
 
+    @Value("${ABANDON_RATIO:0}")
+    private double abandonRatio;
+
+    @Value("${NO_DISPOSITION_RATIO:0}")
+    private double noDispositionRatio;
+
+    @Value("${LOCK_RENEWAL_NEEDED_RATIO:0}")
+    private double lockRenewalNeededRatio;
+
+    @Value("${LOCK_DURATION_IN_MS:30000}")
+    private int lockDurationInMs;
+
+    @Value("${AUTO_RENEW_LOCK:true}")
+    private boolean renewLock;
+
+    private byte[] expectedPayload;
+
+    private final AtomicReference<RunResult> runResult = new AtomicReference<>(RunResult.INCONCLUSIVE);
+
+
     @Override
-    public void run() {
-        ServiceBusProcessorClient processor = getProcessorBuilder(options)
-            .maxAutoLockRenewDuration(Duration.ofSeconds(processMessageDurationMaxInSeconds + 1))
+    public RunResult run() throws InterruptedException {
+        expectedPayload = createMessagePayload(options.getMessageSize());
+
+        ServiceBusProcessorClient processor = toClose(getProcessorBuilder(options)
+            .maxAutoLockRenewDuration(renewLock ? Duration.ofMinutes(5) : Duration.ZERO)
             .maxConcurrentCalls(maxConcurrentCalls)
             .prefetchCount(prefetchCount)
             .processMessage(this::process)
             .processError(err -> {
-                throw LOGGER.logExceptionAsError(new RuntimeException(err.getException()));
+                LOGGER.atError()
+                    .addKeyValue("source", err.getErrorSource())
+                    .log("processor error", err.getException());
+                runResult.set(RunResult.ERROR);
             })
-            .buildProcessorClient();
-
+            .buildProcessorClient());
         processor.start();
-        blockingWait(Duration.ofMinutes(testDurationInMinutes));
-        processor.close();
+
+        blockingWait(options.getTestDuration());
+
+        int activeMessages = getRemainingQueueMessages();
+        for (int extraMinutes = 0; extraMinutes < 3 && activeMessages > 0; extraMinutes++) {
+            blockingWait(Duration.ofMinutes(1));
+            activeMessages = getRemainingQueueMessages();
+        }
+
+        processor.stop();
+        if (activeMessages != 0) {
+            runResult.set(RunResult.WARNING);
+        }
+
+        return runResult.get();
     }
 
     private void process(ServiceBusReceivedMessageContext messageContext) {
+        ServiceBusReceivedMessage message = messageContext.getMessage();
+        if (checkMessage(message)) {
+            blockingWait(Duration.ofMillis(getWaitTime()));
+            settleMessage(messageContext);
+        }
+    }
+
+    private int getWaitTime() {
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        if (random.nextDouble(1) < lockRenewalNeededRatio) {
+            return lockDurationInMs + 1000;
+        } else if (processMessageDurationAvgInMs != 0) {
+            return random.nextInt(processMessageDurationAvgInMs * 2);
```

Comment:
just curious, why are we choosing a value in the range [0, processMessageDurationAvgInMs * 2] instead of [0, processMessageDurationAvgInMs ] ?

Reply:
no good reason. I changed the setting to be Max instead of Avg and removed `*2`.